packages/extension-api/src/extension-api.d.ts

```diff
@@ -708,12 +708,9 @@ declare module '@podman-desktop/api' {
     title?: string;
 
     /**
-     * Controls if a cancel button should show to allow the user to
-     * cancel the long running operation.  Note that currently only
-     * `ProgressLocation.Notification` is supporting to show a cancel
-     * button.
+     * Correspond to a cancel token that can be used to cancel the task
      */
-    cancellable?: boolean;
+    cancellableTokenId?: number;
```

Comment:
it should be kept as it is

1. we can't change things like that on public API
2. it's to control a display thing, it's not related to token ids

Reply:
Then how extension can provide a way to cancel themself a task ?

Inside the `task: (progress: Progress<{ message?: string; increment?: number }>, token?: CancellationToken` the CancellationToken, does not allow to cancel the request.

Therefore extensions can only completed it, or throw an error inside.

Reply:
the idea is that it's the framewor/user that can do the cancellation, not the extension directly. The extension monitor the token to check if it has been canceled and then decide to abort if user asked for cancellation

Reply:
Another solution, we add in the Progress object a cancel method, but it makes no sense that the creator of the task do not have controls over it